### PR TITLE
Fix ModuleNotFoundError: No module named 'cmake'

### PR DESCRIPTION
**Pull request by guoyuhong**
### Description

Described by https://github.com/onnx/onnx/issues/5194

### Motivation and Context

Since `python setup.py build` works fine, I compared the `os.environ` and found that `pip install -e .` add a strange `PYTHONPATH` to `os.environ`. Deleting it make the installation working fine.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -219,6 +219,17 @@ def run(self):
                 raise RuntimeError(
                     "-DONNX_DISABLE_EXCEPTIONS=ON option is only available for c++ builds. Python binding require exceptions to be enabled."
                 )
+            if "PYTHONPATH" in os.environ:
+                # To solve problem that the extra PYTHONPATH env variable added by pip.
+                segs = os.environ["PYTHONPATH"].split(":")
```
Comment (xadupre) [inline setup.py:224]: This won't work on Windows.

Comment (guoyuhong) [reply setup.py:224]: @xadupre Does windows has the same problem? Is it OK to skip windows platform using `"PYTHONPATH" in os.environ and sys.platform != "win32"`

Comment (xadupre) [reply setup.py:224]: I assume miniconda works the same way on Windows. You remove every path from PYTHONPATH already added in sys.path so it starts looking cmake in the installed packages first. Is it possible to add a longer comment then to tell this is needed for conda.

Comment (guoyuhong) [reply setup.py]: @xadupre I can get the same error in Windows env. With the new fix, the problem is also fixed in Windows env.
image

With the fix:

image

**Comment by justinchuby** [comment]
Is this tested in all different environments: venv, conda and w/o venv?

**Comment by guoyuhong** [comment]
@justinchuby I have tested this change in following envs:
1. conda env in Linux, MacOS, Windows 
2. virtualenv in Linux and MacOS
3. w/o venv(python installed in system) in Linux and MacOS

**Review by xadupre — APPROVED** [review]

**Pull request merged**